packages/dev/core/src/Materials/Node/nodeMaterial.ts

```diff
@@ -682,7 +682,7 @@ export class NodeMaterial extends PushMaterial {
 
         // If this is a teleport out, we need to reset the connected block
         if (node.isTeleportOut) {
-            const teleportOut = node as any as NodeMaterialTeleportOutBlock;
+            const teleportOut = node as NodeMaterialTeleportOutBlock;

```

Comment:
We should remove this dependency too, which is the only one remaining preventing the removal of everything related to teleport in `NodeMaterial`...

Maybe by simply making the `_resetDualBlocks` protected and overriding it in `TeleportOutBlock`?